data/src/config/server.rs

```diff
@@ -122,12 +125,20 @@ impl Sasl {
     }
 
     fn external_cert(&self) -> Option<&PathBuf> {
-        if let Self::External { cert } = self {
+        if let Self::External { cert, .. } = self {
             Some(cert)
         } else {
             None
         }
     }
+
+    fn external_key(&self) -> Option<&PathBuf> {
+        if let Self::External { key, .. } = self {
+            Some(key)
+        } else {
+            None
+        }
+    }
```

Comment:
Can we make `key` an Option and return cert path here if key is none? This way we preserve backwards compatibility.

Reply:
Hey, sure can do!  Just to make sure I understand correctly:

The function `external_key` is currently already returning an `Option`, so what you're suggesting is to do another fallback check like this?

```diff
- } else {
+ else if let Self::External { cert, .. } = self {
+    return Some(cert) //fallback
+ } else {
      return None
  }
```

I'm currently mostly worried about odd error cases arising from this, when people use the same config from Linux to macOS/Windows and getting weird error messages -- but I don't have any better idea.

Reply:
I've implemented this patch on top of your work and confirmed it:

- Works w/ previous config (only cert defined w/ cert+key combined file)
- Works w/ separate cert + key files / entries in config

I'm not sure how to advise people on this, other than maybe we update our wiki? At least we don't have breakage for users migrating to this newer patched version and now support separately defined cert / keys for "proper" usage, which ideally people migrate to.

Reply:
Ah, sorry for being MIA for some time!  Sounds great!

Yeah, I was wondering whether one could add some more checking, because it'd be a weird error-case to have a config that works on Linux but you use it on macOS and it breaks (with weird TLS errors, too).  Ideally, maybe one could issue a warning for combined files, going forward?